manual/io/index.md

```diff
@@ -56,784 +46,433 @@ root [1] TFile f1("my.root");
-if you use the `Write()` method.
+The methods of a class are not written to the ROOT file, it contains only the persistent data members.
 
-## Streamers
+See also → [Restrictions on types ROOT I/O can handle]({{ '/manual/io_custom_classes/#restrictions-on-types-root-io-can-handle' | relative_url }}).
 
-When talking about "writing an object to a file", what is actually meant is writing the current values of the data members.  The most common way to do this is to decompose (also called serialize) the object into its data elements and then write them to disk. This decomposition is done by a `Streamer`. Each class that is to be stored in a ROOT file requires a `Streamer`, which decomposes the class and "streams" its members into a buffer.
+### Generating streamers
 
-Variables of composite data types such as classes, structures, and arrays can be decomposed in simple types such as longs, shorts, floats, and chars.
+Streamers are C++ functions that are usually created as part of a class's dictionary, see → [I/O of custom classes]({{ 'manual/io_custom_classes/#generating-dictionaries' | relative_url }}).
 
-The methods of the class are not written to the ROOT file, it contains only the persistent data members. To decompose the parent classes, the `Streamer` calls the `Streamer` of the parent classes. It moves up the inheritance tree until it reaches an ancestor with no parent. To serialize the object data members, it calls their `Streamers`. These in turn move up their own inheritance tree and so on. The simple data elements are written directly to the buffer. Finally, the buffer contains all the simple data members of all the classes that make up that particular object. Data members that are references are never stored, it is always the responsibility of the object's constructor to set them correctly.
+`rootcling` parses the class definition and determines how to stream the object in an optimal way, and which streamers need to be invoked for base classes and members.
 
-### Automatically generated streamers
+### Excluding data members from I/O
 
-A streamer typically calls on other `Streamers`: its parent's streamers and data members. This architecture requires that all classes have `Streamers`, because eventually they will be called. To ensure that a class has a `Streamer`, `rootcling` automatically creates one in the `ClassDef` macro defined in `$ROOTSYS/include/Rtypes.h`. `ClassDef` defines several methods for each class, and one of them is the `Streamer`. The automatically generated streamer is complete and can be used as long as no customization is required.
+To prevent a data member from being written to the file, insert a `!` as the first character in a single-line comment (`//`) following the declaration of the member.
 
-The `Event` class is defined in `$ROOTSYS/test/Event.h`.  It inherits from {% include ref class="TObject" %}.
+To accommodate doxygen-style documentation, this annotation can also be written as `///!`. 
```

Comment:
```suggestion
To accommodate doxygen-style documentation, this annotation can also be written as `///<!`. 
```

Reply:
Thanks, does this mean that `///->` should be `///<->`?

Reply:
Looks very weird but I guess so (@couet @Axel-Naumann)?

Reply:
@couet confirmed it's `///<->`, fixed